### PR TITLE
samples: cellular: udp: add more RAI options 

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
* Add `SO_RAI_NO_DATA` and `SO_RAI_ONGOING`
to this sample to showcase their functionality within
the application.

* Add enable of Access Stratum RAI support during
initialization.

* Update documentation to show how to test the RAI
feature.

### CODE REVIEW

**Review by eivindj-nordic — APPROVED** [review]

File: samples/cellular/udp/src/main.c [hunk 1]
```diff
@@ -105,6 +123,20 @@ static int modem_init(void)
...
 	}
 
+#if defined(CONFIG_UDP_RAI_ENABLE)
+	/* Enable Access Stratum RAI support.
+	 * Note: The 1.3.x modem firmware release is certified to be compliant with 3GPP Release 13.
+	 * %REL14FEAT enables selected optional features from 3GPP Release 14. The 3GPP Release 14
+	 * features are not GCF or PTCRB conformance certified by Nordic and must be certified
+	 * by MNO before being used in commercial products.
+	 */
+	err = nrf_modem_at_printf("AT%%REL14FEAT=0,1,0,0,0");
```
Comment (tokangas) [inline samples/cellular/udp/src/main.c:133]: It's a good idea to enable AS-RAI support here, however this will cause an error with the nRF91x1. The reason is that with nRF91x1 Release 14 features are always supported and the `%REL14FEAT` AT command has been removed.

The easiest way to fix this would be to just ignore the return value from `nrf_modem_at_printf()`. Otherwise a compilation or run time check needs to be added to determine if the command should be sent or not. Maybe `CONFIG_SOC_NRF9160` could be used here, although honestly I think those Kconfig options are a bit messed up.

I settled for this kind of check in a couple of places: https://github.com/nrfconnect/sdk-nrf/blob/9f7148ab7efdbbe88cab9d7b48c0191c225a081c/samples/cellular/gnss/src/assistance_minimal.c#L75

Comment (MirkoCovizzi) [reply samples/cellular/udp/src/main.c:133]: Thanks!

File: samples/cellular/udp/src/main.c [hunk 2]
```diff
@@ -105,6 +136,23 @@ static int modem_init(void)
 		return err;
 	}
 
+#if defined(CONFIG_UDP_RAI_ENABLE)
+	/* Enable Access Stratum RAI support.
+	 * Note: The 1.3.x modem firmware release is certified to be compliant with 3GPP Release 13.
+	 * %REL14FEAT enables selected optional features from 3GPP Release 14. The 3GPP Release 14
+	 * features are not GCF or PTCRB conformance certified by Nordic and must be certified
+	 * by MNO before being used in commercial products.
+	 * Available on nRF9160, unavailable on nRF9161 (features supported by default).
```
Comment (tokangas) [inline samples/cellular/udp/src/main.c:145]: I would rephrase this a bit. With nRF9161 the AS-RAI feature support is always enabled and cannot be disabled. Because of this I think saying "by default" is not correct, because there's no way to change the setting like with nRF9160. For example:
```suggestion
	/* Enable Access Stratum RAI support for nRF9160.
	 * Note: The 1.3.x modem firmware release is certified to be compliant with 3GPP Release 13.
	 * %REL14FEAT enables selected optional features from 3GPP Release 14. The 3GPP Release 14
	 * features are not GCF or PTCRB conformance certified by Nordic and must be certified
	 * by MNO before being used in commercial products.
	 * nRF9161 is certified to be compliant with 3GPP Release 14.
```

**Review by tokangas — APPROVED** [review]

File: samples/cellular/udp/README.rst [hunk 3]
```diff
@@ -166,6 +181,28 @@ After programming the sample to your device, test it by performing the following
...
+-------------------
+
+Test the RAI feature by performing the following steps:
+
+1. |connect_kit|
+#. |connect_terminal|
+#. Connect the nRF91 Series DK to the `Power Profiler Kit II (PPK2)`_ and set up for current measurement.
+#. `Install the Power Profiler app`_ in the `nRF Connect for Desktop`_.
+#. Connect the Power Profiler Kit II (PPK2) to the PC using a micro-USB cable and `connect to it using the App <Using the Power Profiler app_>`_.
+#. Enable RAI by setting the :kconfig:option:`CONFIG_UDP_RAI_ENABLE` option to ``y`` in the :file:`prj.conf` configuration file.
```
Comment (divipillai) [inline samples/cellular/udp/README.rst:194]:
```suggestion
#. Enable RAI by setting the :ref:`CONFIG_UDP_RAI_ENABLE <CONFIG_UDP_RAI_ENABLE>` option to ``y`` in the :file:`prj.conf` configuration file.
```

File: samples/cellular/udp/README.rst [hunk 4]
```diff
@@ -166,6 +181,28 @@ After programming the sample to your device, test it by performing the following
...
+#. Update the data upload frequency by setting the :kconfig:option:`CONFIG_UDP_DATA_UPLOAD_FREQUENCY_SECONDS` option to ``30`` in the :file:`prj.conf` configuration file.
+#. Program the sample to the device.
+#. Power on or reset your nRF91 Series DK.
+#. In the `Power Profiler app` choose a one minute time window.
+#. Observe that after some minutes the average power consumption will settle at around 1.7 mA (may vary depending on network conditions).
+#. Disable RAI by setting the :kconfig:option:`CONFIG_UDP_RAI_ENABLE` option to ``n`` in the :file:`prj.conf` configuration file.
```
Comment (divipillai) [inline samples/cellular/udp/README.rst:200]:
```suggestion
#. Disable RAI by setting the :ref:`CONFIG_UDP_RAI_ENABLE <CONFIG_UDP_RAI_ENABLE>` option to ``n`` in the :file:`prj.conf` configuration file.
```

File: samples/cellular/udp/README.rst [hunk 5]
```diff
@@ -166,6 +181,28 @@ After programming the sample to your device, test it by performing the following
...
+
+Test the RAI feature by performing the following steps:
+
+1. |connect_kit|
+#. |connect_terminal|
+#. Connect the nRF91 Series DK to the `Power Profiler Kit II (PPK2)`_ and set up for current measurement.
+#. `Install the Power Profiler app`_ in the `nRF Connect for Desktop`_.
+#. Connect the Power Profiler Kit II (PPK2) to the PC using a micro-USB cable and `connect to it using the App <Using the Power Profiler app_>`_.
+#. Enable RAI by setting the :kconfig:option:`CONFIG_UDP_RAI_ENABLE` option to ``y`` in the :file:`prj.conf` configuration file.
+#. Update the data upload frequency by setting the :kconfig:option:`CONFIG_UDP_DATA_UPLOAD_FREQUENCY_SECONDS` option to ``30`` in the :file:`prj.conf` configuration file.
```
Comment (divipillai) [inline samples/cellular/udp/README.rst:195]:
```suggestion
#. Update the data upload frequency by setting the :ref:`CONFIG_UDP_DATA_UPLOAD_FREQUENCY_SECONDS <CONFIG_UDP_DATA_UPLOAD_FREQUENCY_SECONDS>` option to ``30`` in the :file:`prj.conf` configuration file.
```

File: samples/cellular/udp/README.rst [hunk 6]
```diff
@@ -166,6 +181,28 @@ After programming the sample to your device, test it by performing the following
...
+#. Program the sample to the device.
+#. Power on or reset your nRF91 Series DK.
+#. In the `Power Profiler app` choose a one minute time window.
```
Comment (divipillai) [inline samples/cellular/udp/README.rst:198]:
```suggestion
#. In the Power Profiler app choose a one minute time window.
```

**Review by divipillai — APPROVED** [review]
Some minor suggestions.

File: samples/cellular/udp/src/main.c [hunk 7]
```diff
@@ -22,6 +22,19 @@ static struct k_work_delayable socket_transmission_work;
 
 K_SEM_DEFINE(lte_connected, 0, 1);
 
+static bool is_nrf9160(void)
```
Comment (eivindj-nordic) [inline samples/cellular/udp/src/main.c:25]: You could also use an internal register to check this: `SIPINFO.PARTNO`

File: samples/cellular/udp/src/main.c [hunk 8]
```diff
@@ -43,11 +56,29 @@ static void socket_transmission_work_fn(struct k_work *work)
...
+	}
+#endif
+
 	err = send(client_fd, buffer, sizeof(buffer), 0);
 	if (err < 0) {
 		printk("Failed to transmit UDP packet, error: %d\n", errno);
 	}
 
+#if defined(CONFIG_UDP_RAI_NO_DATA)
+	/* Let the modem know that there will be no upcoming data transmission anymore.
```
Comment (eivindj-nordic) [inline samples/cellular/udp/src/main.c:74]:
```suggestion
	/* Let the modem know that there will be no upcoming data transmission.
```

**Comment by peknis** [comment]
Changelog entry to be removed from this PR and added as a suggestion to https://github.com/nrfconnect/sdk-nrf/pull/12697.
